Overhaul Shared Code Snippets (#8972)
DevExpress/devextreme-documentation: includes/common-code-on-initialized.md

---
##### Angular

    <!-- tab: app.component.html -->
    <dx-{widget-name}
        (onInitialized)="saveInstance($event)"
    ></dx-{widget-name}>

    <!-- tab: app.component.ts -->
    import { Component } from "@angular/core";
    import { type Dx{WidgetName}Types } from 'devextreme-angular/ui/{widget-name}';

>     import dx{WidgetName} from "devextreme/ui/{widget_name}";
> 
>     // ...
>     export class AppComponent {
>         {widgetName}Instance!: dx{WidgetName};
> 
>         saveInstance(e: Dx{WidgetName}Types.InitializedEvent): void {
>             if (e.component)
>                 this.{widgetName}Instance = e.component;
>         }
>     }

Reviewer: ```suggestion
    import { type Dx{WidgetName}Types } from 'devextreme-angular/ui/{widget-name}';

    // ...
    export class AppComponent {
        {widgetName}Instance: Dx{WidgetName}Types.InitializedEvent['component'] | null = null;

        saveInstance(e: Dx{WidgetName}Types.InitializedEvent): void {
            this.{widgetName}Instance = e.component;
        }
    }
```
*based on [demo](https://js.devexpress.com/Angular/Demos/WidgetsGallery/Demo/DataGrid/AIAssistant/FluentBlueLight/)

##### Vue

    <!-- tab: App.vue (Composition API) -->
    <template>
        <div>
            <Dx{WidgetName} @initialized="saveInstance" />
        </div>
    </template>

    <script setup lang="ts">
    import { ref } from 'vue';
    import { Dx{WidgetName}, type Dx{WidgetName}Types } from 'devextreme-vue/{widget-name}';
    import dx{WidgetName} from "devextreme/ui/{widget_name}";

> 
>     const {widgetName}Ref = ref<dx{WidgetName}>();
>  
>     function saveInstance(e: Dx{WidgetName}Types.InitializedEvent): void {
>         if (e.component)
>             {widgetName}Ref.value = e.component;
>     }

Reviewer: ```suggestion
    import { Dx{WidgetName}, type Dx{WidgetName}Types } from 'devextreme-vue/{widget-name}';

    let {widgetName}Instance: Dx{WidgetName}Types.InitializedEvent['component'] | undefined;
 
    function saveInstance(e: Dx{WidgetName}Types.InitializedEvent): void {
        {widgetName}Instance = e.component;
    }
```
*based on [demo](https://js.devexpress.com/Vue/Demos/WidgetsGallery/Demo/DataGrid/AIAssistant/FluentBlueLight/)

    };
    </script>

##### React

    <!-- tab: App.tsx -->
    import { useCallback, useRef } from 'react';

>     import { {WidgetName}, type {WidgetName}Types } from 'devextreme-react/{widget-name}';
>     import dx{WidgetName} from "devextreme/ui/{widget_name}";
> 
>     function App(): JSX.Element {
>         const {widgetName}Ref = useRef<dx{WidgetName}>();
> 
>         const saveInstance = useCallback((e: {WidgetName}Types.InitializedEvent) => {
>             {widgetName}Ref.current = e.component;
>         }, []);

Reviewer: ```suggestion
    import { {WidgetName}, type {WidgetName}Types } from 'devextreme-react/{widget-name}';

    function App(): JSX.Element {
        const {widgetName}Ref = useRef<{WidgetName}Types.InitializedEvent['component'] | null>(null);

        const saveInstance = useCallback((e: {WidgetName}Types.InitializedEvent) => {
            {widgetName}Ref.current = e.component ?? null;
        }, []);
```
*based on [demo](https://js.devexpress.com/React/Demos/WidgetsGallery/Demo/DataGrid/AIAssistant/FluentBlueLight/)

>     import { {WidgetName}, type {WidgetName}Types } from 'devextreme-react/{widget-name}';
>     import dx{WidgetName} from "devextreme/ui/{widget_name}";
> 
>     function App(): JSX.Element {
>         const {widgetName}Ref = useRef<dx{WidgetName}>();
> 
>         const saveInstance = useCallback((e: {WidgetName}Types.InitializedEvent) => {
>             {widgetName}Ref.current = e.component;
>         }, []);

Reviewer: ```suggestion
    import { {WidgetName}, type {WidgetName}Types } from 'devextreme-react/{widget-name}';

    function App(): JSX.Element {
        const {widgetName}Ref = useRef<{WidgetName}Types.InitializedEvent['component'] | null>(null);

        const saveInstance = useCallback((e: {WidgetName}Types.InitializedEvent) => {
            {widgetName}Ref.current = e.component ?? null;
        }, []);
```
*based on [demo](https://js.devexpress.com/React/Demos/WidgetsGallery/Demo/DataGrid/AIAssistant/FluentBlueLight/)
